doc/hooks/versionCheckHook.section.md

```diff
@@ -16,20 +22,67 @@ stdenv.mkDerivation (finalAttrs: {
+| Rebuild after modification             | The hook rebuilds the package when the hook is modified, since it is a phase running during the build time of the package.                                    | The tester does not require rebuilding the package, since it is a derivation.                                                                                                                               |
+| Overhead during package building       | Negligible. Although running during the build time of the package, the hook is lean and executes few commands.                                                | Zero, since the tester is a derivation that runs after package building.                                                                                                                                    |
+| Content-addressed-derivation awareness | Since it runs during the building, the hook does not deal with failures happening after building, like rewritings that happen post installation.              | Since it runs after the building, the tester detects failures at this time.                                                                                                                                 |
+| Execution by OfBorg CI tool            | Yes, since the hook is a phase running during the build time of the package.                                                                                  | OfBorg does run the tester for _directly affected packages only_; transitive dependencies are ignored by default, requiring extra commands like `@ofborg build dependency1.tests dependency2.tests ...`.    |
+| Execution by `nixpkgs-review` tool     | Yes, since the hook is a phase running during the build time of the package.                                                                                  | No, since the tool has no support for executing `passthru.tests`. [^1]                                                                                                                                      |
+| Package breakage awareness             | Loud and clear as soon as the hook is reached, since it is a phase running during the build time of the package.                                              | Requires specific command to be noticed, e.g. `nix-build -A pkg.tests.version`, since it is a derivation dependent on the package                                                                           |
+| Transitive package breakage            | Never ever goes unnoticed by `nixpkgs-review`, since the tool executes the transitive dependencies.                                                           | Since human beings are prone to forget the duty of running passthru tests and `nixpkgs-review` has no support for running it, it certainly goes unnoticed.                                                  |
```

Comment:
In general, it is not very clear from reading the 1st column of these 2 lines, what exactly is being compared. I understand how you wanted to make it more concise then in the table I wrote in my previous review, but it is not exactly clear what does "breakage" mean, in which context? In Nixpkgs PRs? Or in a user's NixOS system? Or in Hydra?

I admit that even in my table the last line could have been a bit clearer, (I hope you will forgive me for demanding better clarity when this is something that everyone will read v.s something only you read when I suggest something for your PR).

Reply:
> but it is not exactly clear what does "breakage" mean

Breakage is any event that triggers the respective tool.

> , in which context? In Nixpkgs PRs? Or in a user's NixOS system? Or in Hydra?

It does not matter who is the one that triggers the event.

`versionCheckHook` is triggered by anyone that tries to build the package containing it, whether it is a `nixos-rebuild  switch` executed by a GitHub Action or a user running commands in a Powershell instance.

The same can be said about `passthru.tests.version` _mutatis mutandis_.

Reply:
> > but it is not exactly clear what does "breakage" mean
> 
> Breakage is any event that triggers the respective tool.

The respective tool is nixpkgs-review or ofborg? Or the package itself?

> > , in which context? In Nixpkgs PRs? Or in a user's NixOS system? Or in Hydra?
> 
> It does not matter who is the one that triggers the event.

I disagree, because in any of the above contexts there are different chances of a package with `versionCheckHook` or a package's `passthru.tests.version` derivation to be built. This distinction is a very important point for comparison, and the first column of these 2 lines don't make it clear to which context you are referring to.
 
> `versionCheckHook` is triggered by anyone that tries to build the package containing it, whether it is a `nixos-rebuild switch` executed by a GitHub Action or a user running commands in a Powershell instance.

Thanks for adding to the list of possible contexts :). To summarize, the possible contexts I see are:

- A user builds locally packages in a modified Nixpkgs checkout.
- `nixpkgs-review` in a PR.
- Someone running `@ofborg build` in a PR.
- A user running `nixos-rebuild`.
- A GitHub action running `nixos-rebuild`, and possibly other checks if configured to do so.